src/transformers/image_utils.py

```diff
@@ -575,6 +581,17 @@ def read_video_opencv(video_path: str, num_frames: Optional[int] = None, fps: Op
         fps (`int`, *optional*):
             Number of frames to sample per second. Should be passed only when `num_frames=None`.
             If not specified and `num_frames==None`, all frames are sampled.
+        sample_indices_fn (`Callable`, *optional*):
+            A callable function that will return indices at which the video should be sampled. If the video has to be loaded using
+            by a different sampling technique than provided by `num_frames` or `fps` arguments, one should provide their own `sample_indices_fn`.
+            If not provided, simple uniformt sampling with fps is performed, otherwise `sample_indices_fn` has priority over other args.
+            The function expects at input the all args along with all kwargs passed to `read_video_opencv` and should output valid
+            indices at which the video should be sampled. For example:
+
+            def sample_indices_fn(num_frames, fps, metadata, **kwargs):
+                # add you sampling logic here ...
+                return np.linspace(start_idx, end_idx, num_frames, dtype=int)
+
```

Comment:
typos/phrasing mostly, to be applied to the other read_video method

```suggestion
        sample_indices_fn (`Callable`, *optional*):
            A callable function that will return indices at which the video should be sampled. If the video has to be loaded
            by a different sampling technique than provided by `num_frames` or `fps` (frames per second) arguments, one should provide their own `sample_indices_fn`.
            If not provided, a simple uniform sampling at the given fps is performed, otherwise `sample_indices_fn` has priority over the other args.
            The function expects at input all the args along with all kwargs passed to `read_video_opencv` and should output valid
            indices at which the video should be sampled. For example:

            def sample_indices_fn(num_frames, fps, metadata, **kwargs):
                # add your sampling logic here ...
                return np.linspace(start_idx, end_idx, num_frames, dtype=int)

```
Also not sure about superseding the arguments with the sample function - would make more sense to raise so as to disallow this entrypoint, wdyt?

Reply:
yeah, we can't make them mutually exclusive all because custom sampling might need same args as other entrypoints

Reply:
ah, I see, ok then!